src/utils/tokens.ts

```diff
@@ -104,6 +105,34 @@ export const getTokenMap = async (
   return tokenMap;
 };
 
+const overrideCoingeckoId = (tokenInfo: TokenInfo): TokenInfo => {
+  let castTokenInfo = tokenInfo as any;
+
+  const coingeckoId = tokenInfo.extensions?.coingeckoId;
+  if (coingeckoId) {
+    switch (coingeckoId) {
+      case "multi-collateral-dai":
+        castTokenInfo.extensions.coingeckoId = "dai";
+        break;
+      case "wrapped-bitcoin":
+        castTokenInfo.extensions.coingeckoId = "bitcoin";
+        break;
+      case "terra-usd":
+        castTokenInfo.extensions.coingeckoId = "terrausd";
+        break;
+    }
+  } else {
+    const tags = tokenInfo.tags ?? [];
+    if (tags.includes("saber-mkt-luna")) {
+      castTokenInfo.extensions.coingeckoId = "terra-luna";
+    } else if (tags.includes("saber-mkt-sol")) {
+      castTokenInfo.extensions.coingeckoId = "solana";
+    }
+  }
+
+  return castTokenInfo as TokenInfo;
+};
```

Comment:
A couple comments on this:
1. `extensions` has the potential to be undefined, so the `else` could potentially not be type-safe.
2. I set up some types in CoinGecko to handle enforcing existence of a `coingeckoId` that can be used here.

Totally understand the `any` route considering accessing the `coingeckoId`, but think we need to do this to meet type safety:
```suggestion
const overrideCoingeckoId = (tokenInfo: TokenInfo): TokenInfo => {
  let updatedTokenInfo: TokenInfoWithCoingeckoId | undefined;
  if (tokenInfoHasCoingeckoId(tokenInfo)) {
    switch (tokenInfo.extensions.coingeckoId) {
      case "multi-collateral-dai":
        updatedTokenInfo = {
          ...tokenInfo,
          extensions: {
            ...tokenInfo.extensions,
            coingeckoId: "dai",
          }
        }
        break;
      case "wrapped-bitcoin":
        updatedTokenInfo = {
          ...tokenInfo,
          extensions: {
            ...tokenInfo.extensions,
            coingeckoId: "bitcoin",
          }
        }
        break;
      case "terra-usd":
        updatedTokenInfo = {
          ...tokenInfo,
          extensions: {
            ...tokenInfo.extensions,
            coingeckoId: "terrausd",
          }
        }
        break;
    }
  } else {
    const tags = tokenInfo.tags ?? [];
    if (tags.includes("saber-mkt-luna")) {
      updatedTokenInfo = {
        ...tokenInfo,
        extensions: {
          ...tokenInfo.extensions,
          coingeckoId: "terra-luna",
        }
      }
    } else if (tags.includes("saber-mkt-sol")) {
      updatedTokenInfo = {
        ...tokenInfo,
        extensions: {
          ...tokenInfo.extensions,
          coingeckoId: "solana",
        }
      }
    }
  }

  return updatedTokenInfo ? updatedTokenInfo : tokenInfo;
};
```

Reply:
This actually reminds me, do we need to remove the `terrausd` override internal to the Coingecko source, or do we need to keep it because of other sources potentially not working?

Reply:
We could probably remove it from Coingecko source if we apply this function to the entire tokenMap. Right now it's only applied to what we get from Saber registry.

Reply:
> We could probably remove it from Coingecko source if we apply this function to the entire tokenMap. Right now it's only applied to what we get from Saber registry.

Let's do this in a follow-up ticket, don't want to hold up releasing this.